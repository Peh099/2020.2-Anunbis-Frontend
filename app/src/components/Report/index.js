import React from 'react';
import styled from 'styled-components';
import Button from '../Button';
import Checkbox from '../Checkbox';
import FeedPopup from '../FeedPopup';
import Form from '../Form';

import { useForm } from "react-hook-form";
import { yupResolver } from '@hookform/resolvers/yup';
import schema from './validations';

export default function Report({ close }) {
    const { register, handleSubmit, errors } = useForm({
        resolver: yupResolver(schema),
    });

    function onSubmit() {
        console.log("Denuncia realizada.");
    }

    return (
        <FeedPopup title="Denuncia" close={close}>
            <Form onSubmit={handleSubmit(onSubmit)}>
                <Form.Field errorMsg={errors.option?.message}>
                    <Options>
                        <p>Motivo da Denuncia:</p>
                        <Checkbox text="Linguagem ofensiva" name="option" register={register}/>
                        <Checkbox text="Comentário preconceituoso" name="option" register={register} />
                        <Checkbox text="Críticas não relacionadas a disciplina" name="option" register={register} />
                        <Checkbox text="Outros" name="option" register={register} />
                    </Options>
                </Form.Field>
                <Details>
                    <p>Detalhe os motivos:</p>
                    <Form.Field errorMsg={errors.details?.message}><TxtArea name="details" ref={register} /></Form.Field>
                </Details>
                <Footer>
                    <Button type="submit" text="DENUNCIAR" backColor="#F44336" />
                </Footer>
            </Form>
        </FeedPopup>
    );
}

const Options = styled.div`
width: 340px;

label{
    margin: 10px;
}
`;

const Footer = styled.div`
    display:flex;
    justify-content: center;

    button {
        font: bold 14px Ubuntu;
        color: #FFFDE7;
        margin: 15px;
    }
`;

const TxtArea = styled.textarea`
    width: 330px;
    height: 70px;
    border-radius: 10px;
    background-color: #FFFDE7;
    resize: none;

    &:focus{
        outline-width: 0;
    }
`

const Details = styled.div``;