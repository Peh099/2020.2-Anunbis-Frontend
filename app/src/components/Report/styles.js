import styled from 'styled-components';

export const Options = styled.div`
  width: 340px;

  label {
    margin: 10px;
    @media (max-width: 460px) {
      font-size: 11px;
      white-space: nowrap;
      text-overflow: ellipsis;
      overflow: hidden;
    }
  }

  @media (max-width: 460px) {
    width: 70vw;
  }
`;

export const Footer = styled.div`
  display: flex;
  justify-content: center;

  button {
    font: bold 14px Ubuntu;
    color: #fffde7;
    margin: 15px;
    @media (max-width: 460px) {
      font-size: 11px;
      width: 20vw;
      white-space: nowrap;
      text-overflow: ellipsis;
      overflow: hidden;
    }
  }
`;

export const TxtArea = styled.textarea`
  width: 330px;
  height: 70px;
  border-radius: 10px;
  background-color: #fffde7;
  resize: none;

  &:focus {
    outline-width: 0;
  }
  @media (max-width: 460px) {
    width: 75vw;
  }
`;

export const Details = styled.div``;
