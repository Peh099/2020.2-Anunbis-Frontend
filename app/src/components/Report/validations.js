/* eslint-disable */
import * as yup from 'yup';

let schema = yup.object().shape({
  offensive: yup.bool(),
  prejudice: yup.bool(),
  unrelated: yup.bool(),
  others: yup.bool(),

  details: yup
    .string()
    .trim()
    .required('Escreva detalhes sobre o motivo da denuncia.')
    .min(10, 'O detalhamento deve conter pelo menos dez caracteres.')
    .max(254, 'O detalhamento deve ter no mÃ¡ximo 254 caracteres.'),
});

schema = schema.test('myCustomCheckboxTest', null, (obj) => {
  if (obj.offensive || obj.prejudice || obj.unrelated || obj.others) {
    return true;
  }

  return new yup.ValidationError(
    'Escolha ao menos um motivo para a denuncia.',
    null,
    'option',
  );
});

export default schema;
