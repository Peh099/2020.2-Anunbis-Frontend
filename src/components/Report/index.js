import styled from 'styled-components';
import FeedPopup from '../FeedPopup';

const Options = styled.div`

  
`;

export default function Report(){
    
    return(
        <FeedPopup title="Denuncia">
            <Options>
                Motivo da Denuncia:
            </Options>
        </FeedPopup>
    );
}
