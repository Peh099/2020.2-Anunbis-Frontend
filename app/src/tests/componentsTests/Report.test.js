import renderer from 'react-test-renderer';
import Report from '../../components/Report';

describe('Snapshot Report component', () => {
  it('matches the snapshot', () => {
    const tree = renderer.create(<Report />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
