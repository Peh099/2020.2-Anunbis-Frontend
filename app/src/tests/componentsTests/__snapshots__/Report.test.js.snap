// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Report component matches the snapshot 1`] = `
<div
  className="sc-hKFxyN eFovYf"
>
  <div
    className="sc-eCApnc foDSAG"
  >
    <div
      className="sc-jSFjdj eKGict"
    >
      Denuncia
    </div>
    <button
      className="sc-bdnxRM jAGuyW"
      radius="0px"
    >
      X
    </button>
  </div>
  <main
    className="sc-gKAaRy bXsuMg"
  >
    <form
      className="sc-pNWdM dYxkWR"
      onSubmit={[Function]}
    >
      <div
        className="sc-iCoGMd hziUSP"
      >
        <div
          className="sc-kEqXSa gBjBKd"
        >
          <p>
            Motivo da Denuncia:
          </p>
          <label
            className="sc-dlnjwi itVsq"
          >
            <input
              name="offensive"
              type="checkbox"
            />
            <span
              className="sc-gtsrHT ewUAES"
            />
            Linguagem ofensiva
          </label>
          <label
            className="sc-dlnjwi itVsq"
          >
            <input
              name="prejudice"
              type="checkbox"
            />
            <span
              className="sc-gtsrHT ewUAES"
            />
            Comentário preconceituoso
          </label>
          <label
            className="sc-dlnjwi itVsq"
          >
            <input
              name="unrelated"
              type="checkbox"
            />
            <span
              className="sc-gtsrHT ewUAES"
            />
            Críticas não relacionadas a disciplina
          </label>
          <label
            className="sc-dlnjwi itVsq"
          >
            <input
              name="others"
              type="checkbox"
            />
            <span
              className="sc-gtsrHT ewUAES"
            />
            Outros
          </label>
        </div>
        <div
          className="sc-fujyAs FaHra error"
        />
      </div>
      <div
        className="sc-dIsUp"
      >
        <p>
          Detalhe os motivos:
        </p>
        <div
          className="sc-iCoGMd hziUSP"
        >
          <textarea
            className="sc-crzoAE idzbNo"
            name="details"
          />
          <div
            className="sc-fujyAs FaHra error"
          />
        </div>
      </div>
      <div
        className="sc-iqAclL iPqpGC"
      >
        <button
          className="sc-bdnxRM KjNCJ"
          disabled={false}
          type="submit"
        >
          DENUNCIAR
        </button>
      </div>
    </form>
  </main>
</div>
`;
